api/v1alpha1/subscription_types.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023 The Kubebb Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type InstallMethod string
+
+const (
+	// InstallMethodAuto means install directly without confirmation after detecting a new version.
+	InstallMethodAuto InstallMethod = "Auto"
+	// InstallMethodManual means installation process requires user's permission to proceed.
+	InstallMethodManual InstallMethod = "Manual"
+)
+
+// SubscriptionSpec defines the desired state of Subscription
+type SubscriptionSpec struct {
+	// ComponentRef is a reference to the Component
+	ComponentRef *corev1.ObjectReference `json:"component"`
+	// RepositoryRef is a reference to the Repository
+	RepositoryRef *corev1.ObjectReference `json:"repository"`
+	// ComponentPlanInstallMethod is the method used to install the component
+	ComponentPlanInstallMethod InstallMethod `json:"componentPlanInstallMethod,omitempty"`
+	// ForceInstallVersion represents the version that is forcibly installed by the user.
+	ForceInstallVersion string `json:"forceInstallVersion,omitempty"`
```

Comment:
Since we want the functionality of `ComponentPlan` and `Subscription` to be orthogonal, a similar field that forces the installation of a certain version should appear in `ComponentPlan`, so remove this field from `Subscription`.